File: solutions/chapter3/problem_3_4.cpp
#include <bits/stdc++.h>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<int> a(N);
    for (int i = 0; i < N; i++) cin >> a.at(i);

    int max_value = a.at(0);
    int min_value = a.at(0);
    for (int i = 0; i < N; i++) {
        if (max_value < a.at(i)) max_value = a.at(i);
        if (min_value > a.at(i)) min_value = a.at(i);
    }
    
    cout << max_value - min_value << endl;
}

File: solutions/chapter3/problem_3_4.py
def main():
    N = int(input())
    a = list(map(int, input().split()))

    max_value = a[0]
    min_value = a[0]

    for x in a:
        if max_value < x:
            max_value = x
        if min_value > x:
            min_value = x
    
    print(max_value - min_value)
    return

if __name__ == '__main__':
    main()